onnxruntime/core/providers/cpu/cpu_execution_provider.cc

```diff
@@ -196,6 +196,9 @@ class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 9, Eye
 class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 9, float, IsNaN);
 class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 9, MLFloat16, IsNaN);
 class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 9, Erf);
+class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 9, Sinh);
+class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 9, Cosh);
+class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 9, Tanh);
```

Comment:
Why the test case for  tanh passed even before this change?

Reply:
Hmm.. It looks like we have Tanh implementation in opset6..  Thanks for pointing this out!

Reply:
> it appears like we have arch specific implementation for tanh through MLAS [#162](https://github.com/Microsoft/onnxruntime/pull/162), based on a specific perf requirements. Do we know of similar requirements for sinh/cosh ?

sinh and cosh have not been implemented today in mlas, so it's ok to use the eigen based implementations for these ops to ensure we've full support for opset9.